text_formatter.go

```diff
@@ -61,7 +65,16 @@ type TextFormatter struct {
 
 func (f *TextFormatter) init(entry *Entry) {
 	if entry.Logger != nil {
-		f.isTerminal = IsTerminal(entry.Logger.Out)
+		f.isTerminal = f.checkIfTerminal(entry.Logger.Out)
+	}
+}
+
+func (f *TextFormatter) checkIfTerminal(w io.Writer) bool {
+	switch v := w.(type) {
+	case *os.File:
```

Comment:
Ah yes, I should've seen that. :-)

One note is that this actually breaks backwards compatibility. `IsTerminal` is unfortunately exported, but we recently changed the signature (didn't realize people relied on it) and it broke things for a few people. I think it's reasonable to _not_ provide this going forwards—just a heads up.